### PR TITLE
Add test framework generation for mgmt sdk

### PR DESCRIPTION
**Pull request by qiaozha**

### CODE REVIEW

**Comment by sarangan12** [comment]
As discussed offline, I will think if there is any alternative to downloading the test-utils package. Else, I will discuss with releasing it to npmjs to the package owner. Will update this issue soon.

**Review by joheredi — COMMENTED** [review]
Are we planning to add karma config to run tests against the browser?

**Comment by qiaozha** [comment]
> Are we planning to add karma config to run tests against the browser?

Yes, but currently I want to make sure tests against node works

**Comment by qiaozha** [comment]
hold on this PR as the public released @azure-tools/test-recorder package has some issues. 
cc @HarshaNalluru

File: test/smoke/generated/graphrbac-data-plane/src/operations/oAuth2PermissionGrantOperations.ts [hunk 1]
```diff
@@ -8,13 +8,13 @@
...
 import { PagedAsyncIterableIterator } from "@azure/core-paging";
-import { OAuth2PermissionGrant } from "../operationsInterfaces";
+import { OAuth2PermissionGrantOperations } from "../operationsInterfaces";
 import * as coreClient from "@azure/core-client";
 import * as Mappers from "../models/mappers";
 import * as Parameters from "../models/parameters";
 import { GraphRbacManagementClientContext } from "../graphRbacManagementClientContext";
 import {
-  OAuth2PermissionGrantDef,
+  OAuth2PermissionGrant,
```
Comment (qiaozha) [inline test/smoke/generated/graphrbac-data-plane/src/operations/oAuth2PermissionGrantOperations.ts:17]: @sarangan12 why we have changes like this here, but the smoke test is passing.

Comment (qiaozha) [reply test/smoke/generated/graphrbac-data-plane/src/operations/oAuth2PermissionGrantOperations.ts:17]: I don't change anything in the swagger

**Comment by qiaozha** [comment]
@joheredi @sarangan12 Could you help review the PR now ? the test recorder issue should have been resolved.

File: test/smoke/generated/graphrbac-data-plane/src/operations/oAuth2PermissionGrantOperations.ts (hunk 1 above)
Comment (sarangan12) [reply test/smoke/generated/graphrbac-data-plane/src/operations/oAuth2PermissionGrantOperations.ts:17]: It is possible that there is a change in the swagger in the master branch.

**Review by sarangan12 — APPROVED** [review]
Looks good. But, I am seeing your comment that the test recorder has some issues. If that is resolved, this PR could be merged,

**Comment by qiaozha** [comment]
> Looks good. But, I am seeing your comment that the test recorder has some issues. If that is resolved, this PR could be merged,

That issue has been resolved.

**Pull request merged**